vertx-service-discovery/src/main/java/io/vertx/servicediscovery/types/EventBusService.java

```diff
@@ -133,6 +133,36 @@ static Record createRecord(String name, String address, Class itf) {
     return null;
   }
 
+  /**
+   * Lookup for a service record and if found, retrieve it and return the service object (used to consume the service).
+   * This is a convenient method to avoid explicit lookup and then retrieval of the service. A filter based on the
+   * request interface is used.
+   *
+   * @param discovery     the service discovery instance
+   * @param itf           the service interface
+   * @param conf          the configuration for message delivery
+   * @param resultHandler the result handler
+   * @param <T>           the service interface
+   * @return {@code null}
+   */
+  @GenIgnore // Java only
+  static <T> T getProxy(ServiceDiscovery discovery, Class<T> itf, JsonObject conf, Handler<AsyncResult<T>> resultHandler) {
+    JsonObject filter = new JsonObject().put("service.interface", itf.getName());
+    discovery.getRecord(filter, ar -> {
+      if (ar.failed()) {
+        resultHandler.handle(Future.failedFuture(ar.cause()));
+      } else {
+        if (ar.result() == null) {
+          resultHandler.handle(Future.failedFuture("Cannot find service matching with " + filter));
+        } else {
+          ServiceReference service = discovery.getReferenceWithConfiguration(ar.result(), conf);
+          resultHandler.handle(Future.succeededFuture(service.get()));
+        }
+      }
+    });
+    return null;
```

Comment:
any reason to return `null`? Should it just use `void` in the signature?

Reply:
Looks like this was intentional (polyglot support) https://github.com/vert-x3/vertx-service-discovery/commit/d0cb17b5337a2bb1d263e9dc74a247a1a862e0e1